### PR TITLE
Make ASR pipeline compliant with Hub spec + add tests

### PR DESCRIPTION
**Pull request by Rocketknight1**
As well as making the ASR pipeline compliant, this PR also refactors out the output test, so we can just import that in future PRs as well.

### CODE REVIEW

**Comment by Rocketknight1** [comment]
cc @LysandreJik @osanseviero @Wauplin - there will be quite a lot of these if I do pipelines individually, maybe I should batch them into PRs with a few pipelines each?

**Review by vignesh1507 — CHANGES_REQUESTED** [review]
I agree with the changes, could you elaborate more about the changes?

**Review by Wauplin — APPROVED** [review]
Thanks @Rocketknight1! 
I'm not against some batching now that the main helpers are implemented. But note that I'm not a `transformers` maintainers so my approval is mehh :smile:

**Comment by LysandreJik** [comment]
@vignesh1507 you've been posting this message on a few PRs; please explain exactly what you mean as it feels like spam. Thank you.

File: src/transformers/pipelines/automatic_speech_recognition.py [hunk 1]
```diff
@@ -310,6 +309,10 @@ def _sanitize_parameters(
 
         forward_params = defaultdict(dict)
         if max_new_tokens is not None:
+            warnings.warn(
+                "`max_new_tokens` is deprecated and will be removed in version 5 of Transformers. To remove this warning, pass `max_new_tokens` as a key inside `generate_kwargs` instead.",
```
Comment (LysandreJik) [inline src/transformers/pipelines/automatic_speech_recognition.py:313]:
```suggestion
                "`max_new_tokens` is deprecated and will be removed in version 4.49 of Transformers. To remove this warning, pass `max_new_tokens` as a key inside `generate_kwargs` instead.",
```

**Review by LysandreJik — APPROVED** [review]
:fire:

**Comment by LysandreJik** [comment]
I don't feel strongly about batching PRs, if it makes things simpler then let's do it yeah

**Comment by Rocketknight1** [comment]
I'll batch simpler/related ones, in that case, which should save time without making it overwhelming to review.

**Pull request merged**